lib/utils/import/text-files/index.js

```diff
@@ -0,0 +1,58 @@
+import { EventEmitter } from 'events';
+import CoreImporter from '../';
+import { endsWith } from 'lodash';
+
+class TextFileImporter extends EventEmitter {
+  constructor(noteBucket, tagBucket) {
+    super();
+    this.noteBucket = noteBucket;
+    this.tagBucket = tagBucket;
+  }
+
+  importTextFiles = filesArray => {
+    const coreImporter = new CoreImporter(this.noteBucket, this.tagBucket);
+    let importedNoteCount = 0;
+
+    if (!filesArray) {
+      this.emit('stats', 'error', 'No text files to import.');
+      return;
+    }
+
+    const importTextFile = file => {
+      if (!file || !endsWith(file.name.toLowerCase(), '.txt')) {
+        return;
+      }
+
+      // Limit file size we will read to 1mb
+      if (file.size > 1000000) {
+        return;
+      }
+
+      const fileReader = new FileReader();
+      fileReader.onload = event => {
+        const noteContent = event.target.result;
+        if (!noteContent) {
+          return;
+        }
+
+        const modifiedTime = file.lastModified / 1000;
+        coreImporter.importNote({
+          content: noteContent,
+          modificationDate: modifiedTime,
+          creationDate: modifiedTime,
+        });
+
+        importedNoteCount++;
+        this.emit('status', 'progress', importedNoteCount);
+      };
+
+      fileReader.readAsText(file);
+    };
+
+    for (const file of filesArray) {
+      importTextFile(file);
+    }
+  };
```

Comment:
Could we emit a `'complete'` status event at the end of this, too?

Reply:
Yeah, this is async so I need to find a way to know when we've reached the end.

Reply:
Added a solution in 2bdc4bfcd9978bb15f310f7d957cb652d4675fad. I'm not in love with it since it may fire early due to the async nature of `FileReader`, but it will probably suffice.

Reply:
Btw this is exactly where Promises come in handy ✨

Simplified example:

```
const importTextFile = file => {
  return new Promise((resolve, reject) => {
    const fileReader = new FileReader();
    fileReader.onload = event => {

      // import note here...

      this.emit('status', 'progress', importedNoteCount);
      resolve();
    };

    fileReader.readAsText(file);
  });
};

Promise.all(filesArray.map(importTextFile))
  .then(() => {
    this.emit('status', 'complete', importedNoteCount);
  });
```

(Not a blocker, the lastFileName strategy is probably good enough and we can promisify it later!)

Reply:
Oh nice! I haven't worked with Promises before. 👍